storm-core/src/jvm/org/apache/storm/localizer/LocalDownloadedResource.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.localizer;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LocalDownloadedResource {
+    private static final Logger LOG = LoggerFactory.getLogger(LocalDownloadedResource.class);
+    private static class NoCancelFuture<T> implements Future<T> {
+        private final Future<T> _wrapped;
+        
+        public NoCancelFuture(Future<T> wrapped) {
+            _wrapped = wrapped;
+        }
+        
+        @Override
+        public boolean cancel(boolean mayInterruptIfRunning) {
+            //cancel not currently supported
```

Comment:
Cancel is difficult to support, but not impossible.  The only value it offers right now would be to possibly stop downloading a jar/blob early if an assignment changed in the middle of the download.  This is rare enough that I didn't feel it was worth the extra effort to do it at this time.